evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -84,6 +88,13 @@ def test_data_splitter(X_y_binary):
     assert isinstance(automl.rankings, pd.DataFrame)
     assert len(automl.results['pipeline_results'][0]["cv_data"]) == cv_folds
 
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', data_splitter=BalancedClassificationDataCVSplit(n_splits=cv_folds), max_iterations=1,
```

Comment:
@chukarsten I left it since it would be nice to know we still support `StratifiedKFold` as a sanity check, but I can remove it if it could be problematic/unnecessary?